### ISSUE TITLE
Possible breaking change between versions 4.5.3 and 4.6.4

### ISSUE THREAD

**Issue by compsri** [tunnelvisionlabs/antlr4cs#248]
Hi,

We're trying to update to Nuget version 4.6.4 and we noticed what appears to be a breaking change between the 2 versions. I've attached a simple grammar to illustrate this.

Test condition: "$body.properties []"

The above condition will throw a parser error with 4.5.3, which is the expected outcome. With 4.6.4, no error is thrown. I'm trying to understand if it really is a breaking change or if I'm missing something else.

Any help is appreciated.

Thanks!!

[Condition.zip](https://github.com/tunnelvisionlabs/antlr4cs/files/1282830/Condition.zip)

**Comment by sharwell** [comment]
This grammar doesn't have any rules that end with EOF, so it's possible that it would suffer from antlr/antlr4#118. However, assuming everything is working correctly I would expect the output to be the following:

* No input symbols are matched
* The resulting parse tree is `(Eof <EOF>)`

**Comment by compsri** [comment]
[ConsoleApplication3.zip](https://github.com/tunnelvisionlabs/antlr4cs/files/1380476/ConsoleApplication3.zip)
Hi,

Thank you for pointing out the issue in the grammar above! Adding EOF did solve the issue with the negative condition. However, moving forward with the actual grammar I'm running into another issue, which is even more puzzling to me.

I've basically extended the grammar from above slightly, and wrote a custom Visitor extending the BaseVisitor. However, somewhere in the auto-generated code for ANTLR, the return value is being lost and visitor.Visit(IParseTree) is returning null. I've debugged through the code and I can see that ConditionParser.EqualityContext.Accept is returning a non-null value to ConditionBaseVisitor.VisitExpression, but the receiving method has null as the return value.

I've attached my sample VisualStudio project here with all the relevant code.

Thanks!
[ConsoleApplication3.zip](https://github.com/tunnelvisionlabs/antlr4cs/files/1380479/ConsoleApplication3.zip)
